collections/max_by.ts

```diff
@@ -91,13 +91,9 @@ export function maxBy<T>(
   array: Iterable<T>,
   selector: (el: T) => Date,
 ): T | undefined;
-export function maxBy<T>(
+export function maxBy<T, S extends (number | string | bigint | Date)>(
   array: Iterable<T>,
-  selector:
-    | ((el: T) => number)
-    | ((el: T) => string)
-    | ((el: T) => bigint)
-    | ((el: T) => Date),
+  selector: (el: T) => S,
```

Comment:
This allows mixed return type 'selector' like `() => Math.random() > 0.5 ? "abc" : 1`, which we don't want to allow.

Reply:
Other signatures below have the same problems

Reply:
Hm... This is true. Closing.